services/tracking.ts

```diff
@@ -106,23 +107,61 @@ function parseCoordinates(
 }
 
 TaskManager.defineTask(LOCATION_TASK_NAME, async ({ data, error }) => {
+  // Handle errors first
   if (error) {
-    console.log(error.message);
-    return;
+    console.error("Error in location task:", error.message);
+    throw error;
   }
-  if (data) {
-    //@ts-ignore
-    const { locations } = data;
-    console.log("New background location: ", locations[0]);
-    console.log("New background location: ", data);
-    let activeStage = await getActiveStage();
-    if (!activeStage) {
-      throw new Error("No active stage set");
-    }
-    await createLocation(
-      activeStage.id,
-      locations[0].coords.latitude,
-      locations[0].coords.longitude,
-    );
+
+  // Proceed if there is data
+  if (!data) {
+    console.warn("No data received in location task.");
+    throw new Error("No data received in location task.");
+  }
+
+  // Extract locations from the data (ignoring TypeScript warning)
+  //@ts-ignore
+  const { locations } = data;
+  console.log("New background location received:", locations[0]);
+
+  // Fetch the active stage
+  const activeStage = await getActiveStage();
+  if (!activeStage) {
+    throw new Error("No active stage set");
+  }
+
+  // Fetch all locations for the active stage
+  const allStageLocations = await getAllLocationsByStageId(activeStage.id);
+
+  const currentLocation = {
+    latitude: locations[0].coords.latitude,
+    longitude: locations[0].coords.longitude,
+  };
+
+  // Add the new location to the database
+  await createLocation(
+    activeStage.id,
+    currentLocation.latitude,
+    currentLocation.longitude,
+  );
+
+  if (allStageLocations.length >= 1) {
+    const latestLocation = {
+      latitude:
+        // @ts-ignore
+        allStageLocations[allStageLocations.length - 1]._raw.latitude,
+      longitude:
+        // @ts-ignore
+        allStageLocations[allStageLocations.length - 1]._raw.longitude,
```

Comment:
Hier hatte Typescript gemeckert, "latitude" würde es in "RawRecord" nicht geben, daher @ts-ignore.

Ich musste auch den Namen des Arrays verkürzen, sonst wäre "latitude" in die nächste Zeile gerutscht und @ts-ignore hätte nicht gegriffen